src/hooks/useChainId.ts

```diff
@@ -7,11 +8,24 @@ import { parsePrefixedAddress } from '@/utils/addresses'
 
 const defaultChainId = IS_PRODUCTION ? chains.eth : chains.gor
 
-export const useChainId = (): string => {
+// Use the location object directly because Next.js's router.query is not available initially
+const getUrlSafeParam = (): string => {
+  if (typeof location === 'undefined') return ''
+  const pathParam = location.pathname.split('/')[1]
+  const safeParam = /[a-z0-9-]+\:0x[a-f0-9]{40}/i.test(pathParam) ? pathParam : ''
```

Comment:
I've fixed the one that had A-Z, but the other ones are slightly different depending on the context, so I'll let them be.

> When loading the app it now only logs one pageview event but its on the wrong chain. In my case it logs görli even though I am on a rinkeby safe.

Fixed this. It was because the `safe` param can also be in the actual query if the redirect is server side (like on CloudFlare).

Reply:
I tried the pattern on [regexr](https://regexr.com/) but without `A-F` and it wouldn't match an address that has uppercase characters but in the app it seems fine. Do you know why?